src/lib.rs

```diff
@@ -131,6 +132,15 @@ pub enum CheckSupportMethod {
     Head,
 }
 
+/// Which URL should be used for subsequent range requests?
+pub enum RangeRequestUrlSource {
+    /// Use the initial request URL
+    Request,
+
+    /// Use the initial response URL
+    Response,
```

Comment:
Is it _ever_ correct to use the response URL?

Reply:
I think only when signed redirects are used this doesnt work. In other cases the extra hops might be avoidable.

Reply:
maybe in the comments we could explain the tradeoff?

Reply:
Yeah I can definitely explain the cases.

Reply:
> That's a good question, is there a RFC about HEAD/GET range requests?

@zanieb The RFC that covers Range requests seems to be section 14 of the HTTP Semantics one: https://datatracker.ietf.org/doc/html/rfc9110#name-range-requests

(I first found https://datatracker.ietf.org/doc/html/rfc7233, but it says that it's obsoleted by RFC9110)

Reply:
> Is it _ever_ correct to use the response URL?

To me it feels like an optimization that I can't find being discussed in the HTTP rfc, and if I'm right about that it sounds like something that maybe shouldn't be the default at least, but something you opt into if you know you'll never hit cases like the Amazon links in your application...

Reply:
Yeah after reading up on this some more I think I agree. @zanieb If you need a quick fix I would be happy to change the default for now without it being configurable.

Reply:
I think if we shipped a change to the default here, I'd probably upgrade us immediately since we still see reports around this.